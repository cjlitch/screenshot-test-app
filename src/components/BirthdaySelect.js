import React, { Component } from 'react';
import Select from 'react-select';
import 'react-select/dist/react-select.css';

class BirthdaySelect extends Component {

    state = {
        selectedOption: '',
    }
  
    handleChange = (selectedOption) => {
        this.setState({ selectedOption });
    }

    render() {
        return (
            <div>
            <label for="day">Day</label>
            <Select
                name="day"
                value={this.state.selectedOption.value}
                onChange={this.handleChange}
                clearable={false}
                options={[
                    { value: 'one', label: 'One' },
                    { value: 'two', label: 'Two' },
                ]} />
            </div>
        );
    }
}

export default BirthdaySelect;
