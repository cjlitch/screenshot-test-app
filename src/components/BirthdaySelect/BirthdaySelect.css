.birthday-select {
  margin-top: 20px;
  max-width: 200px;
}

.birthday-label {
  font-family: 'Spectral SC', serif;
  font-weight: 400;
}