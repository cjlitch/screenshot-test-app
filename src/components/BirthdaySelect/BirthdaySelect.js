import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Select from 'react-select';
import 'react-select/dist/react-select.css';
import styled from 'styled-components';
import styleguide from '../../styleguide';

export default class BirthdaySelect extends Component {
    static propTypes = {
        label: PropTypes.string.isRequired,
        options: PropTypes.array.isRequired,
        onChange: PropTypes.func.isRequired,
    }

    state = {
        selectedOption: '',
    }
  
    handleChange = (selectedOption) => {
        this.setState({ selectedOption });
        this.props.onChange(selectedOption);
    }

    render() {
        return (
            <StyledSelect data-test="birthday-select">
                <Label htmlFor={this.props.label}>
                    {this.props.label}
                </Label>
                <Select
                    name={this.props.label}
                    value={this.state.selectedOption.value}
                    onChange={this.handleChange}
                    clearable={false}
                    options={this.props.options} />
            </StyledSelect>
        );
    }
}

const Label = styled.label`
    font-family: ${styleguide.fonts.spectral};
    color: ${styleguide.colors.grayTundora};
    font-weight: 400;
`;

const StyledSelect = styled.div`
    margin-top: 20px;
    max-width: 200px;`
;
