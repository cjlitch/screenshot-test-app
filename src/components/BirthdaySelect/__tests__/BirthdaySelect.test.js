import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import BirthdaySelect from '../BirthdaySelect';
import Select from 'react-select';

describe('BirthdaySelect', () => {
    it('has a select', () => {
        const wrapper = shallow(
            <BirthdaySelect
                options={[1, 2, 3]}
                label="test"
                onChange={() => {}} />
            );
            
        expect(wrapper.find(Select)).toHaveLength(1);
    });
});


