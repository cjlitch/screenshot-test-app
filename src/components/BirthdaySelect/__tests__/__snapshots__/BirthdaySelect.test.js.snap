// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirthdaySelect matches snapshot 1`] = `
<div
  className="birthday-select"
  data-test="birthday-select"
>
  <label
    className="birthday-label"
    htmlFor="test"
  >
    test
  </label>
  <div
    className="Select is-searchable Select--single"
    style={undefined}
  >
    <div
      className="Select-control"
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <span
        className="Select-multi-value-wrapper"
        id="react-select-2--value"
      >
        <div
          className="Select-placeholder"
        >
          Select...
        </div>
        <div
          className="Select-input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-activedescendant="react-select-2--value"
            aria-describedby={undefined}
            aria-expanded="false"
            aria-haspopup="false"
            aria-label={undefined}
            aria-labelledby={undefined}
            aria-owns=""
            className={undefined}
            id={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            role="combobox"
            style={
              Object {
                "boxSizing": "content-box",
                "width": "5px",
              }
            }
            tabIndex={undefined}
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </span>
      <span
        className="Select-arrow-zone"
        onMouseDown={[Function]}
      >
        <span
          className="Select-arrow"
          onMouseDown={[Function]}
        />
      </span>
    </div>
  </div>
</div>
`;
